tests/test_dragon_installer.py

```diff
@@ -0,0 +1,519 @@
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import io
+import json
+import pathlib
+import random
+import string
+import tarfile
+import textwrap
+import typing as t
+from collections import namedtuple
+
+import pytest
+from github import Github
+from github.GitReleaseAsset import GitReleaseAsset
+from github.Requester import Requester
+
+import smartsim
+from conftest import FileUtils
+from smartsim._core.entrypoints.dragon_install import (
+    check_for_utility,
+    cleanup,
+    expand_archive,
+    filter_assets,
+    install_dragon,
+    install_package,
+    is_crayex_platform,
+    retrieve_asset,
+    retrieve_asset_info,
+)
+from smartsim.error.errors import SmartSimCLIActionCancelled
+
+# The tests in this file belong to the group_a group
+pytestmark = pytest.mark.group_a
+
+
+mock_archive_name = "dragon-0.8-py3.9.4.1-CRAYEX-ac132fe95.tar.gz"
+_git_attr = namedtuple("_git_attr", "value")
+
+
+@pytest.fixture
+def asset_list() -> t.Dict[str, GitReleaseAsset]:
+    requester = Requester(
+        auth=None,
+        base_url="https://github.com",
+        user_agent="mozilla",
+        per_page=10,
+        verify=False,
+        timeout=1,
+        retry=1,
+        pool_size=1,
+    )
+    headers = {"mock-header": "mock-value"}
+    attributes = {"mock-attr": "mock-attr-value"}
+    completed = True
+
+    assets: t.Dict[str, GitReleaseAsset] = {}
+
+    for python_version in ["py3.9", "py3.10", "py3.11"]:
+        for dragon_version in ["dragon-0.8", "dragon-0.9", "dragon-0.10"]:
+            for platform in ["", "CRAYEX-"]:
+
+                asset1 = GitReleaseAsset(requester, headers, attributes, completed)
+
+                asset1_name = mock_archive_name
+                asset1_name = asset1_name.replace("py3.9", python_version)
+                asset1_name = asset1_name.replace("dragon-0.8", dragon_version)
+                asset1_name = asset1_name.replace("CRAYEX-", platform)
```

Comment:
pedantic style thing: consider making `mock_archieve_name` a formatting string 
```py
mock_archive_name = "{}-{}.4.1-{}ac132fe95.tar.gz"
```
and formatting the name
```py
asset1_name = mock_archive_name.format(dragon_version, python_version, platfrom)
```
rather than manually searching and replacing sub-strings

Reply:
Done!